### ISSUE TITLE
User Documentation with Github

### ISSUE THREAD

**Issue by sreenivasulureddysura** [emmett-framework/granian#358]

**Comment by gi0baro** [comment]
@sreenivasulureddysura can you clarify this?

**Comment by iamkhav** [comment]
@sreenivasulureddysura, do you mean "programmatic" as in calling Granian from Python instead of cli?

If so, that's true. This way of using Granian is not documented. One would have to check code to see how https://github.com/emmett-framework/granian/blob/0a85e88ca9b18fa37cfe265bd08f0c5e9c25230f/granian/cli.py#L257-L304

I could add this use-case to README.md. 

BTW, are there any plans to have a dedicated documentation website?

**Comment by gi0baro** [comment]
> @sreenivasulureddysura, do you mean "programmatic" as in calling Granian from Python instead of cli?

Correct, I changed the title and the naming comes from Uvicorn (https://www.uvicorn.org/#running-programmatically).

> I could add this use-case to README.md.
> 
> BTW, are there any plans to have a dedicated documentation website?

IMHO at the moment there's still few content for launching a website; probably relying on markdown files within the GH repo is fine at the moment, in the same way we have those for other projects in the organization like [renoir](https://github.com/emmett-framework/renoir/tree/master/docs).

**Comment by sreenivasulureddysura** [comment]
@gi0baro Let's bring all the markdown files and deploy it using the github page. Mkdocs library will help in the deploy it.  If you are okay with Mkdocs I will submit the PR with the basic structure.

**Comment by gi0baro** [comment]
> @gi0baro Let's bring all the markdown files and deploy it using the github page. Mkdocs library will help in the deploy it. If you are okay with Mkdocs I will submit the PR with the basic structure.

Again, there's no enough content/reasons to publish (yet?) a website about Granian. And if and when that become a need, it would probably be a subdomain of emmett.sh website or even if on its own domain, [built in a similar way](https://github.com/emmett-framework/website).

None of the projects in Emmett organization use Mkdocs; I would avoid having a single project with a different toolset/workflow, and if there's any valid reason to change that organization-wide, we should have that discussion separately from this issue.

Let's start by having documentation in place with just markdown files, then post-pone further considerations at the proper time.

**Comment by sreenivasulureddysura** [comment]
@gi0baro `f emmett.sh website or even if on its own domain, [built in a similar way](https://github.com/emmett-framework/website).` this looks good.  Can we have a sub-domain of the emmett.sh for granian that helps? You have lot of content for Granian Just the document to arrange it .

**Comment by iamkhav** [comment]
> @gi0baro `f emmett.sh website or even if on its own domain, [built in a similar way](https://github.com/emmett-framework/website).` this looks good. Can we have a sub-domain of the emmett.sh for granian that helps? You have lot of content for Granian Just the document to arrange it .

How about as a first (small) step we re-organize the current root md file into multiple ones and link from the root md file (in places where it might be necessary)? :)

It's easier to read and it'll also be easier to transition whenever @gi0baro decides it's time to move the docs to a standalone domain.

FWIW, I think the thoroughness and quality of information in the docs is much more important than them having their own domain. I'll start with making a PR for programmatic usage ☝🏽
